plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -4862,10 +4862,15 @@ public synchronized Network setupvSwitchNetwork(final Connection conn) {
         return null;
     }
 
-    public void shutdownVM(final Connection conn, final VM vm, final String vmName) throws XmlRpcException {
+    public void shutdownVM(final Connection conn, final VM vm, final String vmName, final boolean forcedStop) throws XmlRpcException {
         Task task = null;
         try {
-            task = vm.cleanShutdownAsync(conn);
+            if (forcedStop) {
```

Comment:
@rhtyd Can you put some relevant logging to indicate hard or clean shutdown? This will help in debugging, relevant for all HV handlers.